proposals/unified-show-diff.md

```diff
@@ -0,0 +1,134 @@
+* By default, show all outputs in the workspace.
+
+`diff` implementations will:
+* Support arbitrary number of Git revisions (more than two) as positional
+  arguments to compare.
+* Support `--targets` as specific files for which to compare revisions.
+* Support comparing the last `n` commits (see `dvc exp show -n` and
+  https://github.com/iterative/dvc/issues/5710).
+* Support comparing an arbitrary number of filenames (more than two) in the
+  current workspace to each other (see
+  https://github.com/iterative/dvc/issues/5693).
+* By default, compare all outputs in the workspace to the `HEAD` commit.
+
+The existing arguments available in the current `show` and `diff`
+implementations will need to be reviewed to determine which can be adopted,
+which need to be changed, and which can be dropped.
+
+### Abstractions from Git
+
+In Git, the [diff
+driver](https://git-scm.com/docs/gitattributes#_defining_an_external_diff_driver)
+determines how the diff is calculated between the files. A custom diff driver
+can be configured in `.gitconfig` and used for specified files in
+`.gitattributes`. Each driver takes the same parameters passed by Git. The
+[difftool](https://git-scm.com/docs/git-difftool) is the frontend that shows the
+outputs of the diffs.  
+
+Metrics, plots, images, and other output types each can be thought of as having
+their own different `show`/`diff` driver (`show` and `diff` can probably share a
+driver). There may be a common set of parameters for all drivers, although there
+might be parameters specific to a particular driver (for example, the `template`
+for `dvc plots`). Each driver should return a machine-readable diff.
+
+There could also be multiple difftool frontends to visualize the diff of all
+output types together (for example, all diffs get aggregated into one HTML page.
+Other downstream apps may implement their own difftools. There also might be
+difftools for specific drivers/output types (for example, an image file viewer).
+
+# How We Teach This
+
+One potential confusion is that "outputs" is vague and used elsewhere in docs,
+but it provides enough flexibility that any future output types can be added
+easily.
+
+Another potential confusion is the number of different ways in which users can
+invoke `diff`, which makes it powerful but possibly overloaded. Care should be
+taken when designing and documenting the `diff` command line interface to make
+it explainable.
+
+If support for custom drivers or difftools is prioritized, these concepts will
+need to be explained along with instructions and examples for how to implement
+them.
+
+The primary sections within the docs that will be impacted are (this is not an
+exhaustive list):
+* content/docs/command-reference/metrics
+* content/docs/command-reference/plots
+* content/docs/start/metrics-parameters-plots.md
+
+# Drawbacks
+
+A common abstraction for all output types minimizes the flexibility of each
+output type's `show` and `diff` commands.
```

Comment:
So the key question is are we forcing the generalization too much? Originally the idea was to merge commands that are already basically the same thing but somehow ended up duplicated. Maybe no need to force it, I think (e.g. throw in `show`, throw in mime/types, etc.) at least as a first iteration. Then again ERs are useful for big changes like that so up to you!